src/Umbraco.Core/Configuration/ConfigConnectionString.cs

```diff
@@ -1,16 +1,61 @@
+using System;
+using System.Data.Common;
+
 namespace Umbraco.Core.Configuration
 {
     public class ConfigConnectionString
     {
-        public ConfigConnectionString(string connectionString, string providerName, string name)
+        public ConfigConnectionString(string name, string connectionString, string providerName = null)
         {
+            Name = name ?? throw new ArgumentNullException(nameof(name));
             ConnectionString = connectionString;
-            ProviderName = providerName;
-            Name = name;
+
+            ProviderName = string.IsNullOrEmpty(providerName) ? ParseProvider(connectionString) : providerName;
         }
 
         public string ConnectionString { get; }
         public string ProviderName { get; }
         public string Name { get; }
+
+        private string ParseProvider(string connectionString)
+        {
+            if (string.IsNullOrEmpty(connectionString))
+            {
+                return null;
```

Comment:
If the connection string is null or empty, wouldn't we want to throw the ArgumentNullException in the constructor too? Or can this legitimately happen as part of the install process

Reply:
Null is expected before install :)